STYLE_GUIDE.md

```diff
@@ -0,0 +1,76 @@
+# Introduction
+
+This is the code style guide for the FAST tooling repository. When contributing to this project please follow the standards defined in this guide.
+
+## Files
+
+File and folder naming in general should not include special characters or space. Alphanumeric is preffered, separated by dash or underscore depending on the type of file. Names should be as concise as possible while also being meaningful.
+
+### Organization
+
+Folder structure should be as flat as possible. This is not to say that all files should go into a single folder, folders should exist where they make sense for larger categorizing purposes and file names can be leveraged to supply additional context.
+
+Example:
+```text
+my-component/
+└─ my-component.ts
+└─ my-component.styles.ts
+└─ my-component.template.ts
+└─ my-component.spec.ts
+```
+
+### Scripts
+
+As seen in the above example, script files should be all lowercase and dash separated. If there is need for additional context, marking a test file in the example above, this should be included separating the additional context with a period at the end of the file name.
+
+### Markdown
+
+Markdown files should be all uppercase, separated by an underscore e.g., CODE_OF_CONDUCT.md. This allows them to stand out from the rest of the files as documentation.
+
+## Syntax
+
+Much of the primarily TypeScript syntax is taken care of via `eslint` and prettier. There are times when the linter is unable to determine when naming conventions are broken, these situations are covered in the sections below.
+
+### Functions & Methods
+
+Functions & methods should be camelCase.
+
+Example:
+```ts
+myFunction(): void {
+    // example code
+}
+```
+
+### Web Component Attributes
```

Comment:
A couple notes on attributes that may be helpful to include something on - some of this may seem obvious, but I see these more often than not. :)

- Avoid auxiliary verbs - prefer `selected` over `isSelected`
- Avoid unnecessary prefixing - prefer `collapsed` instead of `accordionCollapsed`. Unless there are *multiple* collapsed things, prefixing as such is redundant

Reply:
Added!